packages/dropdownMenu/components/DropdownSection.tsx

```diff
@@ -1,12 +1,26 @@
 import * as React from "react";
+import { cx } from "@emotion/css";
 import { DropdownMenuItemProps } from "./DropdownMenuItem";
+import { border } from "../../shared/styles/styleUtils";
 
 export interface DropdownSectionProps {
+  sectionIndex?: number;
```

Comment:
❓ Can we make this required?
Otherwise, it will become `key="dropdown-undefined"` very easily.

Reply:
not sure we can require it and not create TS errors all over our app based on how it is used now. I don't love the original implementation. Let me see if I can figure out a way to explain it. Essentially, the way this was written, DropdownMenu would extend the DropdownSection and DropdownMenuItem components with additional styling, but only if they were used as children of `DropdownMenu`. As such, my changes here, I realize now, are not backwards compatible.

Reply:
wrt to the `key` prop though, thanks for pointing this out because now I see it is not necessary at all

Reply:
> wrt to the key prop though, thanks for pointing this out because now I see it is not necessary at all

I actually wanted to suggest that but I thought I may not know enough how that would work together with these clones - great that it can be removed.